adoc/admin-dr-backup.adoc

```diff
@@ -0,0 +1,196 @@
+
+cat << *EOF* > ${MANIFEST} 
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: etcd-backup
+  namespace: kube-system
+  labels:
+    jobgroup: backup
+spec:
+  template:
+    metadata:
+      name: etcd-backup
+      labels:
+        jobgroup: backup
+    spec:
+      containers:
+      - name: etcd-backup
+        image: ${ETCD_IMAGE}
+        env:
+        - name: ETCDCTL_API
+          value: "3"
+        command: ["/bin/sh"]
+        args: ["-c", "etcdctl --endpoints=https://127.0.0.1:2379 --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt --key=/etc/kubernetes/pki/etcd/healthcheck-client.key snapshot save /backup/etcd-snapshot-\$(date +%Y-%m-%d_%H:%M:%S_%Z).db"]
+        volumeMounts:
+        - mountPath: /etc/kubernetes/pki/etcd
+          name: etcd-certs
+          readOnly: true
+        - mountPath: /backup
+          name: etcd-backup
+      restartPolicy: OnFailure
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        operator: Exists
+      hostNetwork: true
+      volumes:
+      - name: etcd-certs
+        hostPath:
+          path: /etc/kubernetes/pki/etcd
+          type: DirectoryOrCreate
+      - name: etcd-backup
+        hostPath:
+          path: ${ETCD_SNAPSHOT}
+          type: Directory
+*EOF*
+
+kubectl create -f ${MANIFEST}
+```
++
+If you are using local hostPath and not using a shared storage device, etcd back will be created to any one of the master nodes. To find the node associated with each etcd backup:
++
+```bash
+kubectl get pods --namespace kube-system --selector=job-name=etcd-backup -o wide
+```
+
+== Schedule ETCD Cluster Backup
+1. Mount external storage device to all master nodes. This is only required if following step is using local hostpath as volume storage.
+2. Create Cronjob.
++
+HINT: Backup size depends on the cluster. Ensure each of the backup has sufficient space. The available size should be more than the database snapshot file. You should also have a retention method to clean up the unneeded snapshot overtime.
++
+```bash
+ls -sh /var/lib/etcd/member/snap/db
+```
++
+Below is the sample manifest bind to local hostpath. Recommend to use other storage methods instead.
++
+Replace <STORAGE_MOUNT_POINT> with directory to store for backup. The directory must exist on every node in cluster.
++
+Replace <IN_CLUSTER_ETCD_IMAGE> with etcd image used in cluster. This can retrive by accessing any one of node in cluster and run `grep image: /etc/kubernetes/manifests/etcd.yaml | awk '{print $2}'`
++
+```
+ETCD_SNAPSHOT="<STORAGE_MOUNT_POINT>/etcd_snapshot"
+ETCD_IMAGE="<IN_CLUSTER_ETCD_IMAGE>"
+
+# SCHEDULE in Cron format. https://crontab.guru/
+SCHEDULE="*/3 * * * *"
+
+# *_HISTORY_LIMIT is the number of maximum history keep in the cluster.
+SUCCESS_HISTORY_LIMIT="3"
+FAILED_HISTORY_LIMIT="3"
+
+MANIFEST="etcd-backup.yaml"
+
+cat << *EOF* > ${MANIFEST}
+apiVersion: batch/v1beta1
```

Comment:
Is CronJob apiVersion `batch/v1` available?

Reply:
Don't think so :-/
```
no matches for kind "CronJob" in version "batch/v1"
```